### PR TITLE
[Graph widget] Add Decimal Opt for Y-Label Display

### PR DESCRIPTION
**Pull request by BerndN**
Adds option to show no decimals or up to 5 for display of Y-Label

### CODE REVIEW

**Comment by montegoulding** [comment]
@BerndN Sorry that you haven't had a reply on this yet. We did discuss this internally last week but nobody has had a chance to get back to you on it. What we agreed would be best would be to move the number formatting to a separate library module so that it can be used by other widgets easily and also so that when/if we implement something like `format <number> with <format> as string` or probably more helpful `format <list> with <format> as string` we can just update the one place and it will work for all. Would you be keen to look at moving the code to a module?

File: extensions/widgets/graph/notes/Add optional decimal to Y-Labels.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Y-Labels can optionally display decimaals
```
Comment (montegoulding) [inline extensions/widgets/graph/notes/Add optional decimal to Y-Labels.md]: @BerndN I think the filename should be something like `feature-decimal-y-labels.md` and it looks like you missed a typo in `decimals` here.

**Comment by BerndN** [comment]
@montegoulding Sorry for the typo, fixed.

As for a module I don't know what that would be. Never did a module. Could you point me to one?

The code for adding decimals to the Graph Widget is probably not production grade for a module. And one would have to decide about the format, something like "numberFormat" or any other.

**Comment by BerndN** [comment]
@montegoulding OK found WidgetUtils module, wouldn't that be the place to do the formatting?
It already does some formatting like stripZeroes and intToString

**Comment by montegoulding** [comment]
@BerndN You can choose. Modules are cheap.

**Comment by BerndN** [comment]
@montegoulding How about merging this version of the Graph widget with decimals (assuming it passes review) and adding format <list> with <format> as string to WidgetUtils module a bit later? 

For one I will be  out of town for a couple of days and this way I would have more time to test an addition to WidgetUtils module and no dependencies for Graph Widget on WidgetUtils module at the moment.

Graph Widget would be using WidgetUtils module format in a later revision.

**Comment by BerndN** [comment]
@montegoulding and anybody of the team. What are we going to do about this?

If decimal formatting should move to a module with something like
format list with format as string
I would appreciate some suggestions for the syntax. 
Right now I use 0 as indicating _no decimals_ and up to 5 for 5 decimals, filling zeroes up to the number of decimals requested. 
Or should it more closely follow the LCS convention allowing for more flexible formatting also of the integer part?
Should it be a separate module? Or part of an existing module?

**Comment by montegoulding** [comment]
@BerndN I've been playing with the graph a bit lately and was wondering about a few different options. One was to have a callback to the script of the widget to format the label. That way it can be completely flexible for currency or whatever you want to do. I haven't looked at the feasibility of this yet though.

**Comment by runrevmark** [comment]
@montegoulding: That won't really work - widgets shouldn't try and call LCS to process things as that can cause re-entrancy problems. If they do have to call LCS it has to be done in a deferred model - send a message, then get script to update the widget. One option is to figure out a 'format' type string which describes how a number should be formatted as a string - including prefixes etc. and precision - that way no callback is required, the numbers are formatted with respect to the format.

i.e. essentially 'numberFormat' but more flexible, as the value is for display only.

**Comment by runrevmark** [comment]
.NET has quite comprehensive format strings for various datatype -> string conversions:
https://msdn.microsoft.com/en-us/library/26etazsy(v=vs.110).aspx

**Comment by montegoulding** [comment]
Thanks @runrevmark it was only a vague notion I had. If the format string can be flexible enough then that seems a good plan.

**Comment by BerndN** [comment]
I think I will close this pull request. It seems preferable to wait for a generic solution of the valueToString problem.

**Pull request closed without merging**